File: backend/app/drivers/ocserv.py
import subprocess


class OCServDriver:

    def __init__(self, ocpasswd="/etc/ocserv/ocpasswd"):
        self.ocpasswd = ocpasswd

    def create_user(self, username: str, password: str):

        cmd = [
            "ocpasswd",
            "-c",
            self.ocpasswd,
            username,
        ]

        process = subprocess.Popen(
            cmd,
            stdin=subprocess.PIPE,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            text=True,
        )

        process.communicate(f"{password}\n{password}\n")

        return process.returncode == 0

    def delete_user(self, username: str):

        cmd = [
            "ocpasswd",
            "-c",
            self.ocpasswd,
            "-d",
            username,
        ]

        result = subprocess.run(cmd)

        return result.returncode == 0

    def lock_user(self, username: str):

        cmd = [
            "ocpasswd",
            "-c",
            self.ocpasswd,
            "-l",
            username,
        ]

        result = subprocess.run(cmd)

        return result.returncode == 0

    def unlock_user(self, username: str):

        cmd = [
            "ocpasswd",
            "-c",
            self.ocpasswd,
            "-u",
            username,
        ]

        result = subprocess.run(cmd)

        return result.returncode == 0

File: app/ocserv.php
<?php


function ocserv_add_user($username,$password)
{

$cmd="echo '$password' | ocpasswd -c /etc/ocserv/ocpasswd $username";

exec($cmd,$output,$result);


return $result===0;

}



function ocserv_delete_user($username)
{

$cmd="ocpasswd -c /etc/ocserv/ocpasswd -d $username";


exec($cmd,$output,$result);


return $result===0;

}



function ocserv_restart()
{

exec("systemctl restart ocserv");

}



?>
